jerry-core/api/jerry-debugger.c

```diff
@@ -116,3 +116,81 @@ jerry_debugger_cleanup (void)
   }
 #endif /* JERRY_DEBUGGER */
 } /* jerry_debugger_cleanup */
+
+/**
+ * Sets whether the engine should wait and run a source.
+ *
+ * @return true - if the client source has been processed
+ *         false - otherwise
+ */
+bool
+jerry_debugger_wait_and_run_client_source (void)
+{
+#ifdef JERRY_DEBUGGER
+  if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED)
+      && !(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_BREAKPOINT_MODE))
+  {
+    JERRY_CONTEXT (debugger_flags) = (uint8_t) (JERRY_CONTEXT (debugger_flags) | JERRY_DEBUGGER_CLIENT_SOURCE_MODE);
+    jerry_debugger_uint8_data_t *client_source_data_p = NULL;
+
+    while (true)
+    {
+      if (jerry_debugger_receive (&client_source_data_p))
+      {
+        if (!(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED))
+        {
+          jmem_heap_free_block (client_source_data_p,
+                                client_source_data_p->string_size + sizeof (jerry_debugger_uint8_data_t));
+          return false;
+        }
+
+        /* The source arrived. */
+        if (!(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CLIENT_SOURCE_MODE))
+        {
+          jerry_char_t *string_p = (jerry_char_t *) (client_source_data_p + 1);
+          size_t name_size = strlen ((const char *) string_p);
+
+          jerry_value_t ret_value = jerry_parse_named_resource (string_p,
+                                                                name_size,
+                                                                (string_p + name_size + 1),
+                                                                (client_source_data_p->string_size - name_size - 1),
+                                                                false);
+
+          if (!jerry_value_has_error_flag (ret_value))
+          {
+            jerry_value_t func_val = ret_value;
+            ret_value = jerry_run (func_val);
+            jerry_release_value (func_val);
```

Comment:
This is a question: I would consider returning with `true` even if `jerry_run` returns with an error, since the `the client source has been processed`. Maybe even if a parse error occures, since the source is successfully received, just it is wrong.